models/comments-model.js

```diff
@@ -0,0 +1,25 @@
+const db = require('../db/connection')
+
+exports.insertComment = ({ username, body, article_id }) => {
+    return db.query(`
+        INSERT INTO comments 
+        (author, body, article_id)
+        VALUES ($1, $2, $3)
+        RETURNING *;`, [username, body, article_id])
+        .then(({ rows }) => {
+            if(rows.length === 0) {
+                
+                return Promise.reject({ status: 400, msg: 'Cannot insert new comment'})
+            }
+            return rows[0];
+        })
+        .catch((err) => {
+            //catch foreign key error
+            if (err.code = '23503'){
+                
+                return Promise.reject({ status: 404, msg: 'article not found'})
+            } 
```

Comment:
You can still throw this to your error handlers - see if the error ends up in your catch block in the controller without the above. If it does you can pass it along using next